### DISCUSSION TITLE
Should VRS be able to support circular sequences

### DISCUSSION THREAD

**Discussion by larrybabb** [ga4gh/vrs#477] [Q&A] [answered]
Currently, our schema does not let us span the origin which is needed for circular genomic sequences. We need to hear from the community on how this might be done and whether this is a significant issue for any adopters.

**Comment by larrybabb** [comment]
From @ahwagner  on PR ga4gh/vrs-python#362 

> A few minor comments included, but my greatest concern is how circular sequences are handled; SequenceLocation requires start <= end, but this is based on sequence linearity assumptions. Should we update the VRS model to allow end < start for circular sequences? If not, how do we plan to handle variants that span the 0 coordinate?

**Comment by larrybabb** [comment]
From @ehclark on PR ga4gh/vrs-python#362

> My own $0.02 on the circular sequence question. And I'm really just considering human mitochondrial variants here.
> 
> Supporting circular reference sequences and allowing locations that span the zero point seems like a solution in search of a problem. My reasons for thinking this:
> 
> ClinVar does not list any variants close to the zero point
> The gnomAD 3.1 variants close to the zero point are all substitutions or not problematic (meaning normalization does not expand them over the zero point)
> The other formats and tools that will be upstream of VRS computation (HGVS, VCF) all treat the mitochondrial chromosome as a linear sequence or do not support variants that span the zero point
> For example, I ran this test in Variant Validator:
> 
> Unable to validate the submitted variant NC_012920.1:m.16568_3del against the GRCh38 assembly. The following warnings were returned:
> 
> The variant positions are valid but we cannot normalize variants spanning the origin of circular reference sequences
> And from a human biology standpoint I would think that in/dels that disrupt the zero point and replication would be highly unlikely to be compatible with life?

**Comment by ahwagner** [comment, marked as answer]
VRS 2.0 supports circular sequences.
